domain/model/state/modelstate.go

```diff
@@ -192,19 +180,29 @@ INSERT INTO model (uuid, controller_uuid, name, type, target_agent_version, clou
 		agentVersion.Valid = true
 	}
 
-	result, err := tx.ExecContext(ctx, stmt,
-		args.UUID,
-		args.ControllerUUID.String(),
-		args.Name,
-		args.Type,
-		agentVersion,
-		args.Cloud,
-		args.CloudType,
-		args.CloudRegion,
-		args.CredentialOwner.Name(),
-		args.CredentialName,
-	)
+	m := dbReadOnlyModel{
+		UUID:               args.UUID.String(),
+		ControllerUUID:     args.ControllerUUID.String(),
+		Name:               args.Name,
+		Type:               args.Type.String(),
+		TargetAgentVersion: agentVersion,
+		Cloud:              args.Cloud,
+		CloudType:          args.CloudType,
+		CloudRegion:        args.CloudRegion,
+		CredentialOwner:    args.CredentialOwner.Name(),
+		CredentialName:     args.CredentialName,
+	}
+
+	stmt, err := preparer.Prepare(`
+INSERT INTO model (*) VALUES ($dbReadOnlyModel.*)
+ON CONFLICT (uuid) DO NOTHING;
```

Comment:
We should remove this `ON CONFLICT` and add a select to see if it already exists.

Reply:
Changed